includes/entities/class-webmention-item.php

```diff
@@ -0,0 +1,304 @@
+			$params[1] = false;
+		}
+		$var = strtolower( substr( $method, 4 ) );
+
+		if ( strncasecmp( $method, 'get', 3 ) === 0 ) {
+			return $this->$var;
+		}
+
+		if ( strncasecmp( $method, 'has', 3 ) === 0 ) {
+			return ! empty( $this->$var );
+		}
+
+		if ( strncasecmp( $method, 'set', 3 ) === 0 ) {
+			$this->$var = current( $params );
+		}
+	}
+
+	public function set( $key, $value ) {
+		call_user_func( array( $this, 'set_' . $key ), $value );
+	}
+
+	/**
+	 * Setter for $this->updated
+	 *
+	 * @param mixed $updated
+	 *
+	 * @return void
+	 */
+	public function set_updated( $updated ) {
+		if ( is_null( $updated ) ) {
+			return;
+		}
+		if ( $updated instanceof DateTimeImmutable ) {
+			$this->updated = $updated;
+		} else {
+			$this->updated = new DateTimeImmutable( $updated );
+		}
+	}
+
+	/**
+	 * Setter for $this->content
+	 *
+	 * @param string $content
+	 *
+	 * @return void
+	 */
+	public function set_content( $content ) {
+		$this->content = webmention_sanitize_html( $content );
+	}
+
+	/**
+	 * Setter for $this->summary
+	 *
+	 * @param mixed $summary
+	 *
+	 * @return void
+	 */
+	public function set_summary( $summary ) {
+		$this->summary = webmention_sanitize_html( $summary );
+	}
+
+	/**
+	 * Setter for $this->published
+	 *
+	 * @param mixed $published
+	 *
+	 * @return void
+	 */
+	public function set_published( $published ) {
+		if ( is_null( $published ) ) {
+			return;
+		}
+		if ( $published instanceof DateTimeImmutable ) {
+			$this->published = $published;
+		} else {
+			$this->published = new DateTimeImmutable( $published );
+		}
+	}
+
+	/**
+	 * Setter for $this->author
+	 *
+	 * @param mixed $author
+	 *
+	 * @return void
+	 */
+	public function set_author( $author ) {
+		if ( is_string( $author ) ) {
+			if ( wp_http_validate_url( $author ) ) {
+				$this->author = array(
+					'type' => 'card',
+					'url'  => $author,
+				);
+			} else {
+				$this->author = array(
+					'type' => 'card',
+					'name' => $author,
+				);
+			}
+		} else {
+			$this->author = $author;
+		}
+	}
+
+	/**
+	 * Transform to commentdata
+	 *
+	 * @return WP_Comment
+	 */
+	public function to_commentdata() {
+
+	}
+
+	/**
+	 * Check if all fields are set
+	 *
+	 * @return boolean
+	 */
+	public function is_complete() {
+		// returns false for now
+		// @todo implement
+		return false;
+	}
+
+	/**
+	 * Returns the representative entry as array
+	 *
+	 * return array;
+	 */
+	public function to_array() {
+		$array = get_object_vars( $this );
+
+		return array_filter( $array );
+	}
+
+	/**
+	 * Returns the representative entry as JF2 data
+	 *
+	 * return string the JF2 JSON
+	 */
+	public function to_json() {
+		return wp_json_encode( $this->to_array() );
+	}
+
+	/**
+	 * Returns the representative entry as a comment array.
+	 *
+	 * return array;
+	 */
+	public function to_commentdata_array() {
+		$published_gmt = $this->published->setTimeZone( 'GMT' );
+		$meta          = array(
+			'mf2_category'    => $this->category,
```

Comment:
is it a good idea to call them `mf2_*`? I know that you are using it in other plugins, but what about the data we set using JSON-LD or OpenGraph? Shouldn't it be a more generic prefix? Or do we really need a prefix?

Reply:
We still haven't settled this, but even though they come from json-ld or ogp, they are mf2/jf2 property names.